vendor/engines/saml_authentication/app/controllers/saml_authentication/sessions_controller.rb

```diff
@@ -2,13 +2,38 @@ module SamlAuthentication
 
   class SessionsController < Devise::SamlSessionsController
 
+    # Some SSO providers are strict and require that the name_identifier matches
+    # what's logged in on their system for SLO.
+    before_action :store_logging_out_username, only: :destroy
+    after_action :store_winning_strategy, only: :create
+
     protected
 
     # Remove recall, so failures redirect back to sign_in page
     def auth_options
       { scope: resource_name }
     end
 
+    def after_sign_out_path_for(_)
+      idp_entity_id = get_idp_entity_id(params)
+      request = OneLogin::RubySaml::Logoutrequest.new
```

Comment:
It's weird to me that we're referencing OneLogin directly here--shouldn't we be getting the SAML provider from settings or something?

Reply:
We're using `devise_saml_authenticatable` which provides the devise adapter. Under the hood, it uses https://github.com/onelogin/ruby-saml which is built by OneLogin. And as opposed to most gems, they've namespaced all their classes with `OneLogin`. So it's not really a reference to OneLogin as a provider, it's just part of the class name.

This particular method is an override of the devise_saml_authenticatable's session controller.